src/results.rs

```diff
@@ -0,0 +1,33 @@
+use std::fmt;
+use std::error::Error;
+
+#[derive(Debug)]
+pub struct ParsingError {
+    desc: String,
```

Comment:
I rather thought about modelling it after old_io::IoError, have a struct with enum inside.